### PR TITLE
Recreate topology after reconnection

### PR DESCRIPTION
**Pull request by reva2**
Fixes ThreeDotsLabs/watermill#315

### CODE REVIEW

**Comment by reva2** [comment]
@m110 can you please review and approve this PR?

**Comment by marlongerson** [comment]
I am also running into this issue where subscription is not re-established on reconnect.

File: pkg/amqp/subscriber.go [hunk 1]
```diff
@@ -238,6 +234,11 @@ func (s *Subscriber) runSubscriber(
 		}
 	}()
 
+	if err = s.prepareConsume(queueName, exchangeName, logFields); err != nil {
+		s.logger.Error("Failed to prepare consume", err, logFields)
+		return
+	}
```
Comment (m110) [inline pkg/amqp/subscriber.go:240]: @reva2 The issue here might be that you won't receive this error on the first subscriber run (it's just logged). Perhaps it makes sense to run `s.prepareConsume` in all these cases?

Comment (m110) [reply pkg/amqp/subscriber.go:240]: Although that might cause duplicated logs so it's also not an elegant solution. 🤔

**Comment by darri89** [comment]
Any movement on this? I would very much love to see this feature.

**Comment by m110** [comment]
See #25 - the issue with this approach is that the initial error would not be returned, but only logged.

**Pull request closed without merging**